### PR TITLE
feat(360): add unified tailcall GraphQL endpoint for publishing posts

### LINKED ISSUE #360: Create a unified GraphQL endpoint to post blogs to multiple publishers
Create a unified GraphQL endpoint that allows us to publish posts to any publisher. 

**Technical Requirements**
- Use Tailcall to generate a unified GraphQL endpoint
- Support Hashnode and Dev.to
- Update publisher logic on the CI to publish using this new endpoint

### PR DESCRIPTION
**Pull request by onyedikachi-david**
/claim #360
Fixes: #360 

- Create Tailcall schema for unified GraphQL endpoint supporting Hashnode and Dev.to
- Implement new devtoPostHandler using unified GraphQL client
  * Add functions for finding, publishing, and updating Dev.to posts
  * Utilize new DevTo mutations and queries
- Implement new hashnodePostHandler using unified GraphQL client
  * Add functions for finding, creating drafts, updating, and publishing Hashnode posts
  * Utilize new Hashnode mutations and queries
- Update publishing logic to use new handlers
- Integrate with existing CI/CD pipeline
  * Add TAILCALL_ENDPOINT as an environment variable in GitHub Actions workflow

BREAKING CHANGE: Publishing now requires a Tailcall endpoint to be configured

This change simplifies our publishing infrastructure by providing a single
GraphQL endpoint for multiple platforms. It improves maintainability and
makes it easier to add support for new platforms in the future.

### CODE REVIEW

**Comment by neo773** [comment]
Hi,

I was already working on this as it was a private assigned issue.
but nevermind.

**Comment by onyedikachi-david** [comment]
> Hi,
> 
> I was already working on this as it was a private assigned issue. but nevermind.

Really, I am sorry, I didn't know please

**Comment by onyedikachi-david** [comment]
Please review @tusharmath

**Comment by webbdays** [comment]
@coderabbitai
please explain what this PR.
thank you ai.

**Comment by onyedikachi-david** [comment]
Hi, @tusharmath The PR is ready for review.

File: publish-externals/tailcall-unified-schema/main.graphql [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+}
+
+type DevToUpdateArticleResponse {
+  id: ID!
+  title: String!
+  description: String
+  url: String!
+  published: Boolean
+  tags: [String!]
+}
```
Comment (tusharmath) [inline publish-externals/tailcall-unified-schema/main.graphql:202]: We want to have a uniform GraphQL schema for all the publishers. Something like 

```graphql
enum Publisher {
  Devto
  Hashnode
  Medium
}

type Publisher {
  publish(publisher: Publisher, id: ID!, article: ArticleInput): ArticleResponse
}
```

**Comment by tusharmath** [comment]
/tip $60

This is not going to be easy to do with tailcall. Thank you for trying!

**Pull request closed without merging**